plugin-server/tests/worker/ingestion/event-pipeline/populateTeamDataStep.test.ts

```diff
@@ -0,0 +1,83 @@
+import { PipelineEvent, Team } from '../../../../src/types'
+import { UUIDT } from '../../../../src/utils/utils'
+import { populateTeamDataStep } from '../../../../src/worker/ingestion/event-pipeline/1-populateTeamDataStep'
+import { TeamManager } from '../../../../src/worker/ingestion/team-manager'
+
+const pipelineEvent: PipelineEvent = {
+    event: '$pageview',
+    properties: { foo: 'bar' },
+    timestamp: '2020-02-23T02:15:00Z',
+    now: '2020-02-23T02:15:00Z',
+    team_id: 2,
+    distinct_id: 'my_id',
+    ip: '127.0.0.1',
+    site_url: 'https://example.com',
+    uuid: new UUIDT().toString(),
+    token: 'token',
+}
+
+const team: Team = {
+    id: 2,
+    uuid: 'af95d312-1a0a-4208-b80f-562ddafc9bcd',
+    organization_id: '66f3f7bf-44e2-45dd-9901-5dbd93744e3a',
+    name: 'testTeam',
+    anonymize_ips: false,
+    api_token: 'token',
+    slack_incoming_webhook: '',
+    session_recording_opt_in: false,
+    ingested_event: true,
+}
+
+let runner: any
+
+beforeEach(() => {
+    const db = { fetchTeamFromToken: jest.fn().mockResolvedValue(team) }
+
+    runner = {
+        nextStep: (...args: any[]) => args,
+        hub: {
+            db,
+            eventsProcessor: {},
```

Comment:
We're needing to do excessive amounts of mocking here - I'm guessing because this code is testing against db.fetchTeamFromToken instead of teamManager?

Suggestion: Just use/mock teamManager here and trust it's tests are sufficient.

Reply:
copied some scaffolding from another test, simplified this